server/src/server.ts

```diff
@@ -141,11 +140,16 @@ export class TcpClient {
 		})
 
 		socket.on('end', () => {
-			this.log('Ended')
+			// This event is called when the other end signals the end of transmission, meaning this client is
+			// still writeable, but no longer readable. In this situation we just want to close the socket.
+			// https://nodejs.org/dist/latest-v18.x/docs/api/net.html#event-end
+			this.kick('Ended')
 		})
 
 		socket.on('timeout', () => {
-			this.warn('Timeout')
+			// As per the docs, the socket needs to be manually closed.
+			// https://nodejs.org/dist/latest-v18.x/docs/api/net.html#event-timeout
+			this.kick('Timed out')
```

Comment:
the client just reconnects right

Reply:
Correct, there's no disconnect packet, so the client has no idea why the server disconnect them, so it'll reconnect after, iirc, about five seconds.